src/cardano_common_plug.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Common code for cracking Cardano 128-byte length legacy secret Keys (a.k.a XPrv).
+ *
+ * This software is Copyright (c) 2022, Pal Dorogi <pal dot dorogi at gmail.com>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * This file takes replicated but common code, shared between the CPU
+ * and the GPU formats, and places it into one common location.
+ *
+ * This file is common for its future use, and currently there isn't an OpenCL format
+ * implementing this feature.
+ *
+ */
+
+#include "cardano_common.h"
+
+struct fmt_tests cardano_tests[] = {
+	//
+	// https://github.com/CardanoSolutions/ByronWalletRecovery/blob/63ac66ee3532b2a1c72a88c0aa726d0f1ce3ed7a/tools/inspect-keystore/index.js#L34
+	// XPrv is BIP32-Ed25519 key, derived from an extended Ed25519 key (64-byte sk + 32-byte pk)
+	// by appending a 32-byte length chaincode to it, and clear and extra bit ( esk[31] &= 0x20).
+	//      +---------------------------------+-----------------------+-----------------------+
+	//      | Extended Private Key (64 bytes) | Public Key (32 bytes) | Chain Code (32 bytes) |
+	//      +---------------------------------+-----------------------+-----------------------+
+	//      <------------ ENCRYPTED ---------->
+	{ "$cardano$b57361ebe335fa171a260fea7d3277579c212dc74fc2a408d6cbd8a6e7a847cab3c44c5fb190705ddd2698f2d5390798893349b4321e7474b1ce06c9d410b3d6055b42d4a95f19cb34b516a160a306c0eaef398e70ea91da450ccb2a7819e95b0102030405060708091011121314151617181920212223242526272829303132", "Secret1234" },
```

Comment:
I didn't mean you'd drop the `$1` component - I meant you'd only insist on the value being 1 in `valid` and then not parse/store it (and don't have that struct field, which you now correctly removed) until later. But OK, let's stay without the field as well.

Reply:
I am reverting it  to $1 as it seems to me much clearer.